src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -466,8 +465,15 @@ class GraphFrame private(
       indexedVertices.select(
         col(ATTR + "." + ID).cast("long").as(LONG_ID), col(ATTR + "." + ID).as(ID), col(ATTR))
     } else {
-      val indexedVertices = zipWithUniqueId(vertices)
-      indexedVertices.select(col("uniq_id").as(LONG_ID), col("row." + ID).as(ID), col("row").as(ATTR))
+      val withLongIds = vertices.select(ID)
+        .distinct()
+        .repartition(1024, col(ID))
```

Comment:
That would be very expensive if the original DataFrame has few partitions. Again, this is just a quick patch to see whether it fixes the problem. We need more work after confirmation.